bpmn-model/src/main/java/io/camunda/zeebe/model/bpmn/validation/zeebe/InclusiveGatewayValidator.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright © 2017 camunda services GmbH (info@camunda.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.camunda.zeebe.model.bpmn.validation.zeebe;
+
+import io.camunda.zeebe.model.bpmn.instance.InclusiveGateway;
+import io.camunda.zeebe.model.bpmn.instance.SequenceFlow;
+import org.camunda.bpm.model.xml.validation.ModelElementValidator;
+import org.camunda.bpm.model.xml.validation.ValidationResultCollector;
+
+public class InclusiveGatewayValidator implements ModelElementValidator<InclusiveGateway> {
+
+  @Override
+  public Class<InclusiveGateway> getElementType() {
+    return InclusiveGateway.class;
+  }
+
+  @Override
+  public void validate(
+      final InclusiveGateway element, final ValidationResultCollector validationResultCollector) {
+
+    final SequenceFlow defaultFlow = element.getDefault();
+
+    if (defaultFlow != null) {
+      if (defaultFlow.getSource() != element) {
+        validationResultCollector.addError(0, "Default flow must start at gateway");
+      }
+    }
+  }
```

Comment:
Why? You mean the forking inclusive gateway must have only one incoming sequence flow? But how to check  it's the forking inclusive gateway?

Reply:
We want to focus this PR on the forking behavior (i.e. taking the outgoing sequence flows). The joining behavior (i.e. deciding whether to activate the inclusive gateway when an incoming sequence flow is taken) was too complicated to implement correctly. It can be supported in the future, but we want to disable it for now. 

We can disable it by rejecting all process models with an inclusive gateway that has multiple incoming sequence flows. We will only accept models with an inclusive gateway if it has a single incoming sequence flow.

As soon as we are able to build the joining behavior for multiple incoming sequence flows correctly (including the pathfinding algorithm we discussed before), then we can remove this validation again.

Does that make sense?